core/src/main/java/org/apache/iceberg/avro/ValueWriters.java

```diff
@@ -492,4 +576,215 @@ protected Object get(IndexedRecord struct, int pos) {
       return struct.get(pos);
     }
   }
+
+  private abstract static class FloatingPointWriter<T extends Comparable<T>>
```

Comment:
After #2464, would this be needed? Or would you just use the floating point classes from that PR?